tremor-script/src/ast/query/raw.rs

```diff
@@ -181,6 +239,274 @@ impl<'script> Upable<'script> for OperatorDeclRaw<'script> {
+                    });
+                    subquery_stmts.push(stream_stmt);
+                }
+
+                subquery_stmts.extend(subquery_decl.raw_stmts.clone());
+                let subq_module = &self.mangle_id(&self.id);
+                helper.module.push(subq_module.clone());
+
+                let decl_params = subquery_decl.params.clone();
+                let stmt_params = up_maybe_params(self.params.clone(), &mut helper)?;
+                let subq_args = self.get_args_map(&decl_params, stmt_params, helper)?;
+
+                for stmt in subquery_stmts {
+                    match stmt {
+                        StmtRaw::ModuleStmt(m) => {
+                            m.define(helper.reg, helper.aggr_reg, &mut vec![], &mut helper)?;
+                        }
+                        StmtRaw::SubqueryStmt(mut s) => {
+                            s.id = self.mangle_id(&s.id);
+                            s.params = SubqueryStmtRaw::inline_maybe_params(
+                                s.params,
+                                &subq_args,
+                                &mut helper,
+                            )?
+                            .into();
+                            s.inline(&mut helper, &mut query_stmts)?;
+                        }
+                        StmtRaw::Operator(mut o) => {
+                            o.id = self.mangle_id(&o.id);
+                            o.params = SubqueryStmtRaw::inline_maybe_params(
+                                o.params,
+                                &subq_args,
+                                &mut helper,
+                            )?
+                            .into();
+                            let mut o = o.up(helper)?;
+                            o.module.push(subq_module.clone());
+                            query_stmts.push(Stmt::Operator(o));
+                        }
+                        StmtRaw::Script(mut s) => {
+                            s.id = self.mangle_id(&s.id);
+                            s.params = SubqueryStmtRaw::inline_maybe_params(
+                                s.params,
+                                &subq_args,
+                                &mut helper,
+                            )?
+                            .into();
+                            let mut s = s.up(helper)?;
+                            s.module.push(subq_module.clone());
+                            query_stmts.push(Stmt::Script(s));
+                        }
+                        StmtRaw::Stream(mut s) => {
+                            s.id = self.mangle_id(&s.id);
+                            query_stmts.push(Stmt::Stream(s.up(helper)?));
+                        }
+                        StmtRaw::Select(mut s) => {
+                            // TTODO Inline args in where, having, groupby clauses
+                            s.from.0.id = Cow::owned(self.mangle_id(&s.from.0.id.to_string()));
```

Comment:
Is this enough to prevent overlap when there are multiple sub queries? especially when they're in modules?

Reply:
this isn't an issue since `custom_subquery` here is not the definition name of the subquery but rather the name given during creating which is enforced to be unique.